src/features/Posts.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable import/extensions */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loading: boolean;
+  error: string;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loading: false,
+  error: '',
+};
+
+export const init = createAsyncThunk('posts/fetch', (userId: number) => {
+  return getUserPosts(userId);
+});
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder.addCase(init.pending, state => {
+      state.loading = true;
+    });
+
+    builder.addCase(init.fulfilled, (state, action) => {
+      state.posts = action.payload;
+      state.loading = false;
+    });
+
+    builder.addCase(init.rejected, state => {
+      state.loading = true;
```

Comment:
Looks like not fixed
![2024-03-26_11h09_40](https://github.com/mate-academy/react_redux-list-of-posts/assets/61904995/90014e73-e42c-4d3d-ac3c-59f92da4e439)